### PR TITLE
Hard fail on py37

### PR DESCRIPTION
**Pull request by ocefpaf**
Also on this PR:

- consolidate test dependencies;
- test tarball creation and installation.

### CODE REVIEW

**Comment by ocefpaf** [comment]
I'll prepare the PyPI release and conda-package after this one is merged.

File: .travis.yml [hunk 1]
```diff
@@ -10,23 +10,18 @@ python:
...
   - pip install -r requirements.txt
   - pip install -r test-requirements.txt
-  - pip install pytest coverage codecov
-  - pip install -e .
-  # Need to clean up the git repository or pytest gets confused by the extra
-  # cruft that python setup.py install leaves behind
-#  - git clean -xfd
+
+install:
+  - python setup.py sdist && version=$(python setup.py --version) && pushd dist && pip install depfinder-${version}.tar.gz && popd
```
Comment (ericdill) [inline .travis.yml]: what's the advantage of doing this all on one line?

Comment (ocefpaf) [reply .travis.yml]: This creates the tarball and install from it, that way we can catch errors that may happen due to a bad `MANIFEST.in` or other packaging errors. This is not too important and I can remove it if you don't like it.

Comment (ericdill) [reply .travis.yml]: no it doesn't bother me at all. I was mostly just wondering the advantage of all the `&&` to string a bunch of bash together.

Comment (ocefpaf) [reply .travis.yml]: Yeah. IMO `pip` is broken for not checking those when installing from source and not a tarball. I believe this is "fixed" with the `pyproject.toml` b/c you can have `MANIFEST.in`, dependencies, etc all in a single file that is interpreted by `pip`. Not sure if that is working as expected though b/c I never used it.

**Pull request merged**